### PR TITLE
fix: link C# definitions with later initializations

### PR DESCRIPTION
**Pull request by emjin**
C# is like Java in that fields can be defined at the top of a class. Support this.

Test plan: see added tests

### CODE REVIEW

File: tests/patterns/csharp/decl_before_init.cs
Comment (amchiclet) [inline tests/patterns/csharp/decl_before_init.cs:1]: Optional and very nit-picky, sorry 🙏 

What are your thoughts on removing the `using` statements and the mvc controller stuff and just make it a simple class to reduce noise when somebody is reading this test?

`Something1` could also be renamed to `FieldMemberAsCalleeMatches`, and `Something2` maybe `LocalVariableAsCalleeMatches`.

**Review by amchiclet — APPROVED** [review]

File: tests/patterns/csharp/decl_before_init.cs
Comment (amchiclet) [reply tests/patterns/csharp/decl_before_init.cs:1]: (Just realized that you took the example from the Linear link too, so very very optional.)

Comment (emjin) [reply tests/patterns/csharp/decl_before_init.cs:1]: Out of laziness I won't fix this but thanks for flagging!

**Review by aryx — COMMENTED** [review]
yep, looks good.

**Pull request merged**